app/services/cache_service.rb

```diff
@@ -33,8 +34,13 @@ def cache_only_updated_listings
         l['Id'] == fresh_listing['Id']
       end
 
-      cache_single_listing(fresh_listing) unless
-        listing_unchanged?(prev_cached_listing, fresh_listing)
+      cache_single_listing(fresh_listing) unless listing_unchanged?(prev_cached_listing,
```

Comment:
After testing some different options, the evidence so far shows that `process_listing_images` will not get called correctly. My guess is that somewhere else is refreshing the Salesforce cache before we set `prev_cached_listings`. I will spend the day trying a few other things but I think we should move forward with:

- Calling `cache_single_listing` only if unchanged.
- Call `process_listing_images` every time

Reply:
Logging for a0W4U00000IgshdUAB: 
![Uploading Screenshot 2023-09-26 at 10.32.28 AM.png…]()

Reply:
No we have checks in the MultipleListingImage service that check if we need to update before processing